node_modules/col-activities/lib/notifications/daily.js

```diff
@@ -0,0 +1,562 @@
+ *
+ * @param  {Course}                                 course        The course for which to send out the daily notifications
+ * @param  {(AssetActivity|WhiteboardActivity)[]}   activities    The set of activities for which to generate an email subject
+ * @param  {User}                                   currentUser   The user to whom the email will be sent
+ * @return {String}                                               An appropriate email subject
+ * @api private
+ */
+var getSubject = function(course, activities, currentUser) {
+  var subject = null;
+  if (activities.length === 1) {
+    var activity = activities[0];
+    if (activity.type === 'asset_comment') {
+      var commenters = _.pluck(activity.asset.comments, 'user');
+      var summaryActors = getSummaryActors(commenters, currentUser);
+      subject = summaryActors + ' commented on your asset ‘' + activity.asset.title + '’';
+    } else if (activity.type === 'asset_comment_reply') {
+      var commenters = _.pluck(activity.asset.comments, 'user');
+      var summaryActors = getSummaryActors(commenters, currentUser);
+      var comment = activity.asset.comments[0].body;
+      if (comment.length > 40) {
+        comment = comment.substring(0, 40) + '...';
+      }
+      subject = summaryActors + ' replied to your comment ‘' + comment + '’';
+    } else if (activity.type === 'whiteboard_chat') {
+      var commenters = _.pluck(activity.whiteboard.chats, 'user');
+      var summaryActors = getSummaryActors(commenters, currentUser);
+      subject = summaryActors + ' commented on your whiteboard ‘' + activity.whiteboard.title + '’';
+    }
+
+  // Default to a generic subject title if there's more than 1 activity
+  // If the canvas instance for this course has no name, we change the subject slightly as the
+  // "from name" already contains "Asset Library and Whiteboards"
+  } else {
+    subject = 'Today\'s activity in the Asset Library and Whiteboards';
+    if (!course.canvas.name) {
+      subject = 'Today\'s activity'
+    }
+  }
+
+  return subject;
+};
+
+/* Activity aggregation */
+
+/**
+ * Get the relevant activities for a user
+ *
+ * @param  {Object}       data    The relevant course data
+ * @param  {User}         user    The user for which to get the activities
+ * @return {Activity[]}           A set of activities for which the user can be notified
+ * @api private
+ */
+var getActivitiesForUser = function(data, user) {
+  // This variable will contain all activities that can be emailed to the user
+  var emailActivities = [];
+
+  // Asset comments
+  _.each(data.assets, function(asset) {
+    // Whether the user owns the asset
+    var isOwner = asset.users[user.id];
+
+    // All comments made by other users
+    var otherComments = _.filter(asset.comments, function(comment) {
+      return (comment.user_id !== user.id);
+    });
+
+    // All replies made to the current user
+    var replies = _.filter(asset.comments, function(comment) {
```

Comment:
I don't fully understand why we need to distinguish between `replies` and `otherReplies `. The only difference is that `replies` will contain the user's replies to their own comments, which I think is something that can always be excluded from the emails.

Therefore, I think we can just do 

```
var replies = _.filter(asset.comments, function(comment) {
  return (comment.parent && comment.parent.user_id === user.id && comment.user_id !== user.id);
});
```

and work with that